etp-backend/src/main/clj/solita/etp/service/laskutus.clj

```diff
@@ -47,6 +48,13 @@
 (defn find-kuukauden-laskutus [db]
   (laskutus-db/select-kuukauden-laskutus db))
 
+(defn mark-as-laskutettu! [db laskutus]
```

Comment:
mikä on in lausekkeen max raja psql:ssä?

Reply:
tämä miltei kannattaisi laittaa long[] parametrina ks. https://jdbc.postgresql.org/documentation/head/arrays.html

Reply:
where ehtokin yksinkertaistuisi: WHERE id = ANY (:laskutettavat);
ja tämän voisi kirjoittaa yesql:llä

Reply:
muutenkin tämän voisi tehdä yesql:llä koska jesql tukee in-listoja.

Reply:
kannattaisiko vaan tehdä nämä yksi kerrallaan sitä mukaa kun nuo muutkin laskukohtaiset operaatiot tehdään? Yksi tuommoinen UPDATE on aika kevyt asia verrattuna niihin SFTP-asiointeihin ja vastaaviin, niin ei minusta näytä että suorituskyvyn kannalta olisi merkittävää etua tehdä näin.

Reply:
No, näköjään myös nuo sftp:t tehdään koko pömpsille kerralla. Siinä tapauksessa tuo niputtaminen myös psql:n suuntaan varmaan on huomattavasti nopeampi.

Reply:
Yksi tapa voisi olla myös vaan käyttää samaa ehtoa millä todistukset haetaan. Ei pitäisi olla mahdollista, että edelliselle kuulle ilmestyy allekirjoitettuja todistuksia kesken laskutuksen.

Mikäköhän näistä vaihtoehdoista olisi paras?

Reply:
Jos haluaaa vähemmän frameworkkia ja kuitenkin nopeamman kuin rivi kerrallaan sisään, niin tuolla https://clojuredocs.org/clojure.core/partition voisi pilkkoa kohtuullisen kokoisiksi viipaleiksi.

Reply:
yleensä sql lausekkeessa ei ole tarkoitus olla hirveän isoa määrää parametreja. psql:n limiittiä en kuitenkaan tunne; oraclessa tämä oli 1000. Lähtökohtaisesti itse käyttäisin array-tyyppiä kaikissa isoissa listoissa jo pelkästään siksi että sql on paljon yksinkertaisemman näköinen, mutta tuota pitäisi testata että mikä toimii parhaiten.

Reply:
myös tuon poimintaehdon käyttäminen pitäisi olla ihan toimiva ratkaisu. Väliin ei pitäisi tulla enää uusia et:tä. Luulisin että z-aikaan ja aws-kelloihin voi tuon verran luottaa että aika ei mene taaksepäin tunneilla.